src/gui/processing/qgsprocessingmultipleselectiondialog.cpp

```diff
@@ -322,14 +326,27 @@ QgsProcessingMultipleInputPanelWidget::QgsProcessingMultipleInputPanelWidget( co
   connect( addDirButton, &QPushButton::clicked, this, &QgsProcessingMultipleInputPanelWidget::addDirectory );
   buttonBox()->addButton( addDirButton, QDialogButtonBox::ActionRole );
   setAcceptDrops( true );
+
+  // populate already selected layer values
+  QgsProcessingContext context;
+  context.setProject( project );
+  for ( const QVariant option : std::as_const( selectedOptions ) )
+  {
+    QIcon icon = QIcon();
+    QgsMapLayer *layer = QgsProcessingUtils::mapLayerFromString( option.toString(), context, false );
+    if ( layer )
+      icon = QgsIconUtils::iconForLayer( layer );
+
+    addOption( option, mValueFormatter( option ), true, false, icon );
+  }
```

Comment:
I don't particularly like this approach -- it's expensive, involving opening layers (potentially there are 1000s), only to throw them away immediately. I'd rather see:

- the logic moved to QgsProcessingMultipleInputPanelWidget::populateFromProject, so that we check if there's an existing item we added for the existing selection which matches that layer but doesn't have a icon set, and set it there.
- or, failing that, just no icons for existing selection